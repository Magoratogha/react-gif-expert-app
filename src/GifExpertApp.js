import { Fragment, useState } from "react";
import AddCategory from "./components/AddCategory";
import GifsGrid from "./components/GifsGrid";

function GifExpertApp() {

    const [categories, setCategories] = useState(['One Punch Man']);

    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <AddCategory submitCallbackFn={ setCategories }/>
            <hr></hr>
            <ol>
                { categories.map((category) => 
                    <GifsGrid key={category} category={ category } />) }
            </ol>
        </Fragment>
    );
}

export default GifExpertApp;