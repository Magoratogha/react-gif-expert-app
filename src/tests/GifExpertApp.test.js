import { render } from "@testing-library/react";
import GifExpertApp from "../GifExpertApp";

describe('Test in <GifExpertApp />', () => {

    test('should render the component properly', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });
});