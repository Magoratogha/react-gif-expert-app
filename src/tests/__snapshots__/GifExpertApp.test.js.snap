// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifExpertApp /> should render the component properly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <form>
    <input
      type="text"
      value=""
    />
  </form>
  <hr />
  <ol>
    <h3>
      One Punch Man
    </h3>
    <p
      class="animate__animated animate__flash"
    >
      Loading...
    </p>
    <div
      class="cards-grid"
    />
  </ol>
</div>
`;
